### PR TITLE
chore: code style

### LINKED ISSUE #155: rm top level code from test files
see for instance https://github.com/maelle/swamp

### PR DESCRIPTION
**Pull request by dpprdan**
This cleans up the code and uses a consistent code style. 
It add documentation for internal functions.
It adds the styler and roxygenise/document workflows. 
Moves test objects to helper-opencage.R, closes #155

### CODE REVIEW

File: tests/testthat/helper-opencage.R [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    condition = oc_key_present(),
+    message = "OpenCage API key is missing"
+  )
+}
+
+# test data ---------------------------------------------------------------
+
+## forward -----------------------------------------------------------------
+
+oc_locs <- c("Nantes", "Flensburg", "Los Angeles")
```
Comment (maelle) [inline tests/testthat/helper-opencage.R]:
```suggestion
oc_locs <- function() {
 c("Nantes", "Flensburg", "Los Angeles")
 }
```

File: tests/testthat/test-oc_forward.R [hunk 2]
```diff
@@ -1,36 +1,12 @@
...
+# Test oc_forward functions -----------------------------------------------
 
 # oc_forward --------------------------------------------------------------
 
 test_that("oc_forward works", {
   skip_if_no_key()
   skip_if_oc_offline()
 
-  res1 <- oc_forward(locations)
+  res1 <- oc_forward(oc_locs)
```
Comment (maelle) [inline tests/testthat/test-oc_forward.R]:
```suggestion
  res1 <- oc_forward(oc_locs())
```

Comment (dpprdan) [reply tests/testthat/test-oc_forward.R]: I don't see that much of a difference between using objects or functions for this. Am I missing something? 🤔 
But the upside of being indifferent is that either way is fine. 😉

Comment (dpprdan) [reply tests/testthat/test-oc_forward.R]: uh, meant to reply [here](https://github.com/ropensci/opencage/issues/155#issuecomment-1385014022)...

**Pull request merged**